library/Test/Hspec/JUnit/Formatter.hs

```diff
@@ -0,0 +1,93 @@
+-- | An Hspec formatter that produces a JUnit XML file of results
+--
+-- @
+-- -- file test/SpecHook.hs
+-- module SpecHook where
+--
+-- import Test.Hspec
+-- import Test.Hspec.JUnit.Config
+-- import Test.Hspec.JUnit.Formatter qualified as Formatter
+--
+-- -- | To only produce a JUnit file, silencing other output
+-- hook :: Spec -> Spec
+-- hook = Formatter.use defaultJUnitConfig
+--
+-- -- | To produce a JUnit file /in addition/ to the normal output
```

Comment:
The whole reason I'm doing this, by the way. So we can get the test output back in our CI logs even when we're also writing a JUnit report.